modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/MustBeVariantAnalyzer.cs

```diff
@@ -49,9 +48,10 @@ private void AnalyzeNode(SyntaxNodeAnalysisContext context)
                     continue;
 
                 var typeSymbol = sm.GetSymbolInfo(typeSyntax).Symbol as ITypeSymbol;
-                Debug.Assert(typeSymbol != null);
+                Helper.ThrowWhenNull(typeSymbol);
 
-                var parentSymbol = sm.GetSymbolInfo(parentSyntax).Symbol;
+                var parentSymbol = sm.GetSymbolInfo(parentSyntax!).Symbol;
```

Comment:
Shouldn't need this null forgiveness operator since we're checking the nullability of `parentSyntax` earlier.

```suggestion
                var parentSymbol = sm.GetSymbolInfo(parentSyntax).Symbol;
```